### PR TITLE
Proposal for an implementation of a recordBone

### PR DESCRIPTION
**Pull request by phorward**
The recordBone can be used to store a data structure defined via a RelSkel within a skeleton. This enables to build more complex building blocks within a skeleton without adding a new module for that.

### CODE REVIEW

**Comment by tsteinruecken** [comment]
I would like to defer this one to 2.5 or 3.0, just from looking at the code there are too many open questions regarding this one (Representation for searching/ordering/filtering in the datastore? What's the point of having this bone with multiple=False? How to integrate #71? What about inter-bone dependencies / Skeleton validation)

File: bones/recordBone.py [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+			cleanList = []
+
+			for item in tmpList:
+				err = self.isInvalid(item)
+				if err:
+					errorDict["%s.%s" % (name, tmpList.index(item))] = err
+				else:
+					cleanList.append(item)
+
+			if not cleanList:
```
Comment (sveneberth) [inline bones/recordBone.py]: In a customer project we needed to change this line to:
```python
if not cleanList and self.required:
```
This allow us to call fromClient with an empty dict, wich means we will clean this bone and the bone has no value.

**Comment by tsteinruecken** [comment]
@phorward Can you run an extensive test on this? I've added the necessary constraints and removed code that handled the now forbidden corner-cases. Also i improved handling of non-required bones.
